system/lib/fetch/asmfs.cpp

```diff
@@ -1939,6 +1940,23 @@ long __syscall146(int which, ...) // writev
   return total_write_amount;
 }
 
+__wasi_errno_t __wasi_fd_write(
+    __wasi_fd_t fd,
+    const __wasi_ciovec_t *iovs,
+    size_t iovs_len,
+    size_t *nwritten
+)
+{
+  long result;
+  result = __syscall146(146 /*writev*/, fd, iovs, iovs_len);
```

Comment:
Doesn't this change completely from syscall 146?  Or is this just local to this file?    Perhaps worth fully removing it from there too?

Reply:
ASMFS implements syscall146 earlier in this file. This PR basically provides a compatibility layer between the two. I did it this way because I didn't want to do a more serious refactoring of the ASMFS code.

If we intend to do major work to productionize ASMFS then I think it may make sense to remove the ASMFS syscall146 code, but not sure we've had any discussions on that.

Reply:
Can you at least add a TODO here to replace __syscall146 completely with __wasi_fd_write.   Given that they are so similar it should be fairly easy I think?   Having refernces to __syscall146 stick around after this change is somewhat confusing but I don't want to block this change necessarily.

Reply:
Good idea, thanks, I added a TODO now.